### DISCUSSION TITLE
TPM-RS alignment meetings.

### DISCUSSION THREAD

**Discussion by ThomasOD** [tpm-rs/tpm-rs#179] [Organization]
Hi guys,

During the Kirkland F2F BoF there was talk about having future alignment meetings (virtual) with a monthly cadence; perhaps starting from November. I just wondered if anyone is planning to do this?
I think it would be a good idea to discuss the SW architecture and perhaps coordinate work tasks so we don't start working on the same features :-)

**Comment by chrisfenner** [comment]
I think this is a great idea. @ThomasOD would you like to propose a time?

**Comment by chrisfenner** [comment]
Revisiting this based on some informal discussions with Microsoft folks --

@trent-reed @oddcoder @reynoldsbd @Christian-Alfonso 

Would it make sense to try to set up a ~monthly sync just to touch base with each other? What time zone is everyone in? I assume most of us are in US Pacific time but wouldn't expect all of us to be.

**Reply by silviuk** [reply]
Hi all, everyone on Huawei side is in Central Europe time, 9h ahead of Pacific. After some internal alignment we would propose to have the call at 5pm CET/8am PT on either Monday, Tuesday or possibly Friday, every 4 weeks or so.

**Reply by chrisfenner** [reply]
This works for me personally, I could do either Monday or Tuesday at that time.

**Reply by oddcoder** [reply]
Would it be a  good idea to have the first meeting next week (first week of October)? I am not available the week after.

**Reply by ghost** [reply]
I'm fine with these dates/times. Thanks!

**Reply by silviuk** [reply]
Given it's "next week" already, I assume today or tomorrow is too short notice. So should we fix the first call for Oct 14, 5pm CET/8am PT?

As to the agenda - besides discussing issues currently under development, from our side we'd be glad to learn more about the roadmap you guys have in mind about the project.

**Reply by TRJensen** [reply]
Timeslot "Monday Oct 14, 5pm CET" is fine with me 👍 🙂

**Reply by ghost** [reply]
That time sound good to me, thanks!

**Comment by oddcoder** [comment]
I am based in Europe (CET/CEST). So current timezone is UTC+2

**Comment by oddcoder** [comment]
So I created with doc for the meeting https://hackmd.io/@oddcoder/ryjHhaqAA
Right now the document is empty, but I invite everyone to fill in anything they would like to discuss. The document is readable/writable by anyone, if that turns to be a problem then we can restrict access.

I will not be available during most of this week, but I will try to fill some points towards the end of the week!

**Comment by chrisfenner** [comment]
Has somebody already volunteered to take care of setting up some kind of video call?

We may be able to request TCG Admin to create a Zoom call for this, if not.

**Comment by silviuk** [comment]
Having a TCG Zoom call would probably be best. I can write to Admin
tomorrow, or feel free to do so if you have time before.

**Reply by chrisfenner** [reply]
Go right ahead, thank you!

**Reply by silviuk** [reply]
TCG Admin just sent the meeting link through TPM WG, please check your calendars as I'm not sure if it's ok to share here. @trent-reed I assume @chrisfenner can share it with you.

Talk soon!

**Reply by silviuk** [reply]
@chrisfenner it seems we need you to start the Zoom meeting as host

**Reply by chrisfenner** [reply]
Oh no! OK, starting the meeting momentarily. Sorry.

**Reply by silviuk** [reply]
Seems someone joined the call as host, but doesn't allow us in. We'll drop in 5 min and we can reschedule later.

**Reply by chrisfenner** [reply]
Uhoh. I also cannot start the meeting. At this point, I think we should reschedule. Sorry, all.

**Reply by silviuk** [reply]
we can also use the Huawei system here (join from browser):
https://app.huawei.com/wmeeting/join/97633509/NV8LX3wRrnFqJGRzmkFdq0eHxMcc7Fw9U

**Reply by silviuk** [reply]
For us @oddcoder @TRJensen @ThomasOD @silviuk tomorrow at same time would work. We can either use the link above, or have another one set up by TCG Admin.

**Reply by ghost** [reply]
I'm fine with tomorrow at the same time.

**Reply by silviuk** [reply]
ok let's hope this works for Chris as well. If so, let us know which meeting link you prefer to use :)

**Reply by chrisfenner** [reply]
I can do tomorrow. I can join via the TCG Zoom link.

**Reply by silviuk** [reply]
Great, then let's try again tomorrow :)

**Comment by oddcoder** [comment]
@ThomasOD @silviuk @chrisfenner @TRJensen @trent-reed  

I was made aware that there are still some technical difficulties with setting up zoom link. In particular it is not clear who would have host rights. To avoid being in the same situation as yesterday, I would like to invite you all to join the meeting today at 5:00pm CET / 8:00 AM PT using Huawei meeting website 

https://app.huawei.com/wmeeting/join/94754539/Cge66Cv87apVFDgwTCFanvPaTuCWD2auZ

And the meeting notes are hosted here: https://hackmd.io/@oddcoder/ryjHhaqAA

**Reply by ghost** [reply]
Chris and I are currently in the Zoom call, are you unable to join? I think the issue was with the original link.

**Comment by oddcoder** [comment]
Hello @ThomasOD  @TRJensen  @trent-reed  @chrisfenner @bradlitterell, kindly find the hackmd document for the next meeting on 11th here, feel free to add/update: https://hackmd.io/@oddcoder/rJHmdlY-1e/edit

@chrisfenner feel free to provide a meeting link.

**Reply by chrisfenner** [reply]
Meeting link: https://us06web.zoom.us/j/82359538490?pwd=HbiJ8IlgnbSozRd9IxUyilSf4BER15.1

To all: Please make sure your username includes your full name and company, so we can let you in from the waiting room.

**Comment by bradlitterell** [comment]
Is this the 11th (Monday) or Tuesday (the 12th)?

I have a conflict on Monday morning.  I will be able to call in, but only for the first 30 minutes.

**Reply by oddcoder** [reply]
> I have a conflict on Monday morning.  I will be able to call in, but only for the first 30 minutes.

It was actually Monday 11th, fixed in the original document. Thanks for  pointing it out.

**Comment by oddcoder** [comment]
Do we have a calendar item / google group for managing the event (or equivalent)? @chrisfenner

**Reply by TRJensen** [reply]
@chrisfenner Have you had time to create a Zoom meeting link for todays meeting? 🙂👍

**Reply by chrisfenner** [reply]
> @chrisfenner Have you had time to create a Zoom meeting link for todays meeting? 🙂👍

Thank you for the reminder! We should be able to join via this Zoom link:

https://us06web.zoom.us/j/82585191888?pwd=lCEnbLQMP23PV0El9JNa9Ltc0a3Tli.1

**Comment by chrisfenner** [comment]
January has been a bit hectic. Should we meet in February? Here are some possible times:

- Monday, February 10, 8AM PST / 5PM CET
- Tuesday, February 11, 8AM PST / 5PM CET
- During the TCG Members Meeting (February 17-20)
- Monday, February 24, 8AM PST / 5PM CET
- Tuesday, February 25, 8AM PST / 5PM CET

@bradlitterell @silviuk @TRJensen @trent-reed

**Comment by bradlitterell** [comment]
In favor of finding a time at TCG or the 10th

**Reply by ghost** [reply]
Any of those work for me, the only days I can't meet on are Fridays.

**Reply by oddcoder** [reply]
Same, I have no preference.

**Reply by TRJensen** [reply]
Hi - I am unavailable in the week 7 (10-16 February). The 3 last options works for me 👍🙂🎉

**Reply by silviuk** [reply]
All dates fine with me.

**Reply by chrisfenner** [reply]
Well, it's already the 10th. There may be a little time during TCG next week but I am already a bit overbooked for the week, so I don't want to overcommit.

I propose Monday, February 24 for our next meeting.

**Reply by ghost** [reply]
24th works for me, thanks!

**Reply by bradlitterell** [reply]
24th works for me, 8am pacific works for me

**Comment by oddcoder** [comment]
Works for me as well but only after 18:00 CET since I have the day booked
before.

**Comment by oddcoder** [comment]
I wont be able to join at 8 due to work related conflicts would 9 be Ok for
you?

**Comment by bradlitterell** [comment]
I can do 9

**Comment by chrisfenner** [comment]
Hi all, sorry for the lateness of setting this up. Our Zoom link is

https://us06web.zoom.us/j/84503450177?pwd=5CPacmavJFi8YVoFcHOk6yshdYxQTk.1

**Comment by chrisfenner** [comment]
I've updated the Zoom call to recur on the 4th Monday of each month at the same time. Happy to tweak it but I think it's helpful to have the zoom call all set up ahead of time.

**Comment by bradlitterell** [comment]
thanks

-Brad

***@***.***?anonymous&ep=bwmEmailSignature> Time to meet with ***@***.***?anonymous&ep=bwmEmailSignature>

**Comment by oddcoder** [comment]
@chrisfenner is there a callender item ? I missed last meeting partly because I didn't have it in calendar.  I prefer not to miss one tomorrow.

If it possible, will it be possible to have zoom invitations ? and if so would you please upload it here?
 or if you prefer to send it to us, maybe `araoof at amazon dot com`
 
 
Alternatively if you want to worry about that, I can take care of it, but we do not use zoom, we use "aws chime". And ummm to describe it in one sentence, all the advertised pieces are there, and it does work. but maybe zoom with calendar would be better.

**Reply by chrisfenner** [reply]
The Zoom call is now recurring, so I think we just need to "calendar-ize" it.

Does this link work:

ICS:
https://us06web.zoom.us/meeting/tZAod-qsqzsrHdPxmjupn7dmELadLPbA72pd/ics?meetingMasterEventId=QAKKtQ5qSuGue14iXl2p0w

Or this one:

Google Calendar:
https://us06web.zoom.us/meeting/tZAod-qsqzsrHdPxmjupn7dmELadLPbA72pd/calendar/google/add?meetingMasterEventId=QAKKtQ5qSuGue14iXl2p0w

**Reply by oddcoder** [reply]
First one says `Invalid meeting ID. (3,001)` second one says `Looks like you can't access this page right now. (200)`

**Reply by chrisfenner** [reply]
OK. Let me create a (public) Google Calendar that can have this event attached to it. One moment...

**Reply by chrisfenner** [reply]
Can you see this?

https://calendar.google.com/calendar/u/0?cid=Y18xYTU0NGRkNGJmYzc3NTgxY2UzYTYxMjI3MTE0YjhlZTlmNzFjODhjZWU5MWZhMTdlNTFlZWRkOWM5OGViMWU3QGdyb3VwLmNhbGVuZGFyLmdvb2dsZS5jb20

**Reply by oddcoder** [reply]
Yup that one works, thanks for making that happen. Maybe we should place this somewhere visible to everyone like github.com/tpm-rs

**Reply by chrisfenner** [reply]
Sounds good, let's let a couple others chime in on if this solution works for them (in case a shared google calendar is not a great solution for everyone) and then publicize

**Reply by silviuk** [reply]
For the call today: from Huawei side we don't have anything to report due to no bandwidth in March.

Does anyone else have topics to discuss?

**Reply by chrisfenner** [reply]
We might not have many. I have a hard stop at 30 minutes in, we could meet but cut the meeting short?

**Reply by oddcoder** [reply]
>  I have a hard stop at 30 minutes in

Alright 30 minutes then, just to make sure we are all aligned meeting is at 17:00 CET

**Reply by oddcoder** [reply]
I think I am stuck outside the meeting "waiting for host to start the meeting"

**Comment by chrisfenner** [comment]
Sorry for the delay, I've just started the meeting (I think)

**Comment by bradlitterell** [comment]
@chrisfenner the links in the thread above, I can't access in Zoom.  One gives me invalid meeting id, the other says I have no access.

**Reply by chrisfenner** [reply]
Does this work?

https://www.google.com/url?q=https://us06web.zoom.us/j/84503450177?pwd%3D5CPacmavJFi8YVoFcHOk6yshdYxQTk.1

**Reply by silviuk** [reply]
I plan to join with 5-8 min of delay, apologies.

**Comment by chrisfenner** [comment]
@bradlitterell @trent-reed @silviuk @ThomasOD @oddcoder @TRJensen 

Our next scheduled meeting is on a US holiday. Should we reschedule? The following Monday is during the TCG Members Meeting.

**Reply by TRJensen** [reply]
I have no update this month, don't know about others. Perhaps we skip this one, and meet at regular time next month 👍

**Reply by chrisfenner** [reply]
That would work for me.

**Reply by ghost** [reply]
Sorry I didn't respond, but since nobody veto'd it, I assume we're all on the same page anyways. I'm totally fine skipping this meeting.

**Comment by josephlr** [comment]
Hey everyone! Reminder that next meeting is Monday July 28th at 9am Pacific / 5pm CET.

Zoom Link:  https://us06web.zoom.us/j/84503450177?pwd=5CPacmavJFi8YVoFcHOk6yshdYxQTk.1

I updated the notes for the upcoming call (https://hackmd.io/@oddcoder/ryjHhaqAA) with some tentative things to talk about, feel free to add anything. See you Monday!

**Reply by silviuk** [reply]
Minor remark: it's 6pm CET

**Comment by josephlr** [comment]
Hey everyone! Reminder that next meeting is Monday August 25th at 9am Pacific / 6pm CET.

Zoom Link: https://us06web.zoom.us/j/84503450177?pwd=5CPacmavJFi8YVoFcHOk6yshdYxQTk.1

I updated the meeting notes (https://hackmd.io/@oddcoder/ryjHhaqAA) to include @jarkkojs's ideas in https://github.com/tpm-rs/tpm-rs/issues/196. Feel free to add any additional topics.

**Reply by jarkkojs** [reply]
I'm happy to join!

**Reply by jarkkojs** [reply]
Unfortunately I cannot participate (outside work conflict). You know the basics, maybe it is good to mention that the repo is now here: https://git.kernel.org/pub/scm/linux/kernel/git/jarkko/tpm2-protocol.git

And I have request put out for mailing list at vger.kernel.org. I'm interested to join in future tho.

**Reply by josephlr** [reply]
@jarkkojs sounds good! We'll be sure to discuss what development workflows would work the best for folks working on this project. 

And thanks for getting tpm2-protocol into its own repository, we will be sure to go through it during the meeting.

**Reply by jarkkojs** [reply]
Yeah, as said this is just a personal life conflict :-) Happy to join next time.

**Reply by jarkkojs** [reply]
https://lore.kernel.org/tpm2/aKzaTYCI2GO_UPRB@kernel.org/T/#u

We will use already familiar to many tpm2@lists.linux.dev for patches. If that cause problems/complains,
only then makes sense to open a separate list but as of today it's so low traffic that it's a good way to start doing stuff.

As per pretty printing (which is not included in the main crate) I can work on the topic if I have time and perhaps we can discuss about that +1 meeting. I've put this a few places but this give a better idea what it can do.

![tpm2sh](https://github.com/user-attachments/assets/17502df8-23e1-4344-9c80-aa1246944adf)

**Reply by josephlr** [reply]
~~@jarkkojs it seems like that above link is broken (at least for me). It looks like lore.kernel.org has a new thing called "Anubis", which just gives me "invalid response". Is there an alternative way to get what you linked above?~~

Edit: nevermind I was able to fix it. For others potentially running into this issue, I was able to resolve it by first visiting https://lore.kernel.org/ and https://lore.kernel.org/tpm2/, passing the challenge, then using the above link.

**Reply by jarkkojs** [reply]
I put here also note why I thought that mailing list is the best option.

There's obviously bunch of undiscovered bugs because it would be beyond human capabilities to write 8K lines of code in three weeks without introducing them but it is better to fix them with a slower cycle and proper discussion. The thing with protocols is that even obvious looking fixes have high risk of introducing bugs as side-effects. Starting from next patch I do for tpm2-protocol, I'll post to that list.

**Comment by josephlr** [comment]
Hey everyone! Remainder that the next meeting is Monday, September 29th, 2025 at 9AM Pacific (PDT) / 18:00 Europe (CEST)

Zoom Link: https://us06web.zoom.us/j/84503450177?pwd=5CPacmavJFi8YVoFcHOk6yshdYxQTk.1

I updated the notes doc with a proposed agenda: https://hackmd.io/@oddcoder/ryjHhaqAA (I'll try to get a better system for committing notes to the tpm-rs repo before the meeting)

CC: @jarkkojs @TRJensen @chrisfenner @oddcoder

**Comment by josephlr** [comment]
Hey everyone! Given that many of us just met last week at the TCG members meeting, would it work to postpone the alignment meeting to **next week**? I would also be down to change which day of the week we meet if there's a better day than Monday (we could do Tuesday or Wednesday morning instead).

I'm also fine keeping Monday mornings, in which case our next meeting would be:

Date:  Monday, November 4
Time: 9am - 10am Pacific (PST) / 18:00 Europe (CET)
Zoom Link: https://us06web.zoom.us/j/84503450177?pwd=5CPacmavJFi8YVoFcHOk6yshdYxQTk.1

CC: @jarkkojs @TRJensen @chrisfenner @oddcoder @bradlitterell @trent-reed

**Reply by josephlr** [reply]
And sorry for the late notice; I should have sent this out last week.

**Comment by bradlitterell** [comment]
Mondays are best for me (my Tue/Wed/Thur/Fri mornings are all already booked. ☹)

**Comment by josephlr** [comment]
Given the holiday weekend in the US next week, I think we would reschedule the alignment meeting to the week after Thanksgiving which would be December 1st. We could also do December 8th instead.

Does anyone have a preference between the two dates (December 1st vs 8th)?

CC: @jarkkojs @TRJensen @chrisfenner @oddcoder @bradlitterell @trent-reed @deeglaze

**Reply by bradlitterell** [reply]
Prefer 12/8

**Reply by deeglaze** [reply]
8th. I'll be traveling on the 1st.

**Reply by josephlr** [reply]
Sounds good, we'll do the 8th

**Comment by josephlr** [comment]
Next Alignment meeting will be on December 8th at 9am PST (18:00 CET, 17:00 GMT)

Zoom Link: https://us06web.zoom.us/j/84503450177?pwd=5CPacmavJFi8YVoFcHOk6yshdYxQTk.1

**Reply by jarkkojs** [reply]
I'll put this to my calendar and join for the first time :-) Sorry I've not been participating earlier! I've been just really super busy.

One topic I think should be addressed by the industry overall is TPMKey ASN.1 format [1]. It needs to get into official IETF process and we do need an official well supported cross-platform key format for TPM keys.

@Foxboron has already submitted patches for IETF creation data but I don't know what is the status of those patches TBH.

Linux kernel has rudimentary support for the unofficial ASN.1 format but we do not want to expand that further before "IETF draft happens". It would be just nice that TPM stacks would be able to cross-talk with each other with a common shared data format.

[1] https://www.hansenpartnership.com/draft-bottomley-tpm2-keys.html

**Comment by bradlitterell** [comment]
Can't say I understand the reason for the ASN.1 encoding proposal for TPM keys, particularly in this forum.  The explicit goal of TPM-RS is to implement the existing TPM specifications, not to create modifications to the existing one, or to contribute to IETF.  Those seem more properly the domain of the TPM WG inside TCG.  We can certainly forward the conversation to TPMWG.

Regarding the IETF proposal itself - the reason seems to be the idea that the key can be loaded across devices. IOW, it seems to only meet its stated purpose when the key is specifically an IMPORT blob, not a PRIVATE blob, though the spec describes both.  The TPM spec already describes the binary formats required for import blobs (which can trivially be converted to OCTET_STREAM), so why this spec?

Among other things, I'm concerned that this seems to be creating an extra spec that has to be concretely compatible with TPM2.0 which seems inflexible.  For example, the spec has a flag for "rsaParent" and implicitly assumes there are only RSA and ECC parents.  I don't think that is a permanent, viable assumption as crypto evolves in a PQC world.

**Reply by jarkkojs** [reply]
Unfortunately was on sick leave today. OK, I won't take this topic to this meeting.

`rsaParent` is one of the most ridiculous attributes in ASN.1 spec TBH as one would get much out of just dumping `TPMT_PUBLIC`. It does not bother me as I just ignore it but it neither makes any sense. The big problem  with that file format is lack of any legit identifier for parent. Before that happens I don't we just increase kernel support.

**Reply by jarkkojs** [reply]
That said, I'll hold on for the IETF and give my feedback over there instead of e.g., this meeting but yep lots of things are just plain wrong in the ASN.1. It should be dead obvious from table 6 of architecture spec that handle is not a valid identifier for parent.

<img width="930" height="543" alt="image" src="https://github.com/user-attachments/assets/ff705e98-3064-4399-8c4d-027cbeff4464" />

**Reply by jarkkojs** [reply]
I think that otherwise the spec makes sense to me. E.g., I've implemented full policy compiler and executor based on the spec (it lacks still TPM2_PolicyAuthorize but does implement TPM2_PolicySecret which tests pretty well the requirements for the former): https://crates.io/crates/tpm2sh

I was doubtful about expections first for e.g., TPM2_PolicySecret but noticed that in a real implementation those expections actualy do make sense. So it is not like I'm opposing everything but some details are just plain objectively wrong, and my soemwhat strong opinions inherit from experience, and seeing what works and what does not. I've spent last 12 years reading TCG specs and made nearly full coverage implementation for the ASN.1 spec.

One detail the spec has no information is TPM2_PolicyOr. in myimplementation I just open code the full TPM2_PolicyRestart jam. Here I don't have strong views because information does not exist.

Additional references:

- https://crates.io/crates/tpm2-policy-language
- https://crates.io/crates/tpm2-tpmkey
- https://crates.io/crates/tpm2-vtpm

**Reply by jarkkojs** [reply]
To not sound like "opposing everything" I will say that policy part is otherwise great except not being explicit how TPM2_PolicyOr is encoded. Now that I have a compiler that generates tpm key with policies, and look at my executor for it in Rust, I can pretty accurately estimate that we could do circa in 500-900'ish lines of C code in-kernel executor for the format. How parents are referenced is exactly and accurately where I'm in full opposition. Aside from that I'm happy.

**Reply by jarkkojs** [reply]
I want to say this. If there's someone reading this and would be interested on implementing policy executor for Linux kernel, I'm happy to mentor on that and provide support (and I have a compiler to generate test data). Contact me: jarkko@kernel.org

**Comment by josephlr** [comment]
Hey everyone, sorry for not having the Google Calendar match the TCG Calendar. I think things got messed up when I changed the date to avoid the Thanksgiving conflict.

I have a conflict for today (Jan 26th), so if it's OK with everyone else, I think it will be best to move the Monthly Sync to next week, and have it be on the first Monday of the Month going forward (let me know if this doesn't work for folks).

If this works, the next meeting would be:

Date: February 2nd, 2026
Time: 9am PST (18:00 CET, 17:00 GMT)
Zoom Link: https://us06web.zoom.us/j/84503450177?pwd=5CPacmavJFi8YVoFcHOk6yshdYxQTk.1
[Google Calendar Link](https://calendar.google.com/calendar/u/0?cid=Y18xYTU0NGRkNGJmYzc3NTgxY2UzYTYxMjI3MTE0YjhlZTlmNzFjODhjZWU5MWZhMTdlNTFlZWRkOWM5OGViMWU3QGdyb3VwLmNhbGVuZGFyLmdvb2dsZS5jb20)
[TPMWG Calendar Link](https://members.trustedcomputinggroup.org/wg/TPMWG/calendar/event/23726/)

CC: @jarkkojs @TRJensen @chrisfenner @oddcoder @bradlitterell @trent-reed @deeglaze @nnooney

**Comment by josephlr** [comment]
Hey Everyone! The next alignment meeting is coming up:

Date: June 1st, 2026
Time: 9am PDT (5pm BST, 6pm CEST, 16:00 UTC)
Zoom Link: https://us06web.zoom.us/j/84503450177?pwd=5CPacmavJFi8YVoFcHOk6yshdYxQTk.1
[Google Calendar Link](https://calendar.google.com/calendar/u/0?cid=Y18xYTU0NGRkNGJmYzc3NTgxY2UzYTYxMjI3MTE0YjhlZTlmNzFjODhjZWU5MWZhMTdlNTFlZWRkOWM5OGViMWU3QGdyb3VwLmNhbGVuZGFyLmdvb2dsZS5jb20)
[TPMWG Calendar Link](https://members.trustedcomputinggroup.org/wg/TPMWG/calendar/event/23726/2026-06-01)

**Reply by josephlr** [reply]
Sorry for the delay, trying to get logged in to the TCG Account to host the Zoom meeting

**Comment by josephlr** [comment]
For the alignment meeting, I was unable to get logged in to the @TCGadmin account, so we are going to switch to Google Meet:

https://meet.google.com/cor-ywyi-kof

**Comment by nnooney** [comment]
Alignment meeting notes are now recorded as discussions in the tpm-rs GitHub organization: https://github.com/orgs/tpm-rs/discussions/categories/meeting-notes?discussions_q=

**Comment by nnooney** [comment]
Closing this discussion, please refer to the individual discussions at https://github.com/orgs/tpm-rs/discussions/categories/meeting-notes?discussions_q=
